src/server/routes/annualData.js

```diff
@@ -69,7 +69,7 @@ function annualCurrentListData(req, res, next) {
       next();
     })
     .catch((error) => {
-      logger.error(`Status Code: ${error.statusCode}, Error Message: ${error.code}`);
+      logger.error(`Status Code: ${error.statusCode}, Error Message: ${error.code}, Source: annualData.annualCurrentListData, Redirecting to: ${config.baseUrl}/404`, error);
```

Comment:
We have `source` here that does helps us locate the function, but we still don't know which endpoint caused the error. The `error` here is not logged out either.

I tired console `error`, and it should have the info as below

```
{ statusCode: 400,
  data: '{"error":"invalid_client","error_description":"The client credentials are invalid"}' }
```

While some useful info we might want to log, unfortunately it won't give us any endpoint info back.

Reply:
I gave the app the wrong credentials to log into Platform API, but I get the following error in log
```
{"timestamp":"2018-05-18T20:57:48.571Z","levelCode":3,"level":"ERROR","pid":"11644","message":"Status Code: 400, Error Message: undefined, Endpoint: /book-lists?type=teens, Redirecting to: /books-music-movies/recommendations/best-books/404"}
```
It says Error Message is undefined. Any pointers?

Reply:
because there is no `error.code` anymore. If you tried to console `error` as I did, the error description is actually nested in `data`.